sql/src/main/java/io/crate/cluster/gracefulstop/DecommissioningService.java

```diff
@@ -304,12 +310,11 @@ public void handle(Signal signal) {
         handle();
     }
 
-    public void handle() {
+    public CompletableFuture<Void> handle() {
         if (dataAvailability == DataAvailability.NONE) {
             exit();
-        } else {
-            decommission();
         }
+        return decommission();
```

Comment:
this modifies the handling logic with decommissioning _always_ executing. Is this intended?

Reply:
I'm not sure it changes much since after `System.exit()`, the decommissioning will not be executed. Am I missing something?

Reply:
ah right it's a System exit inside